### PR TITLE
Use Environment.SystemDirectory to get system directory

### PR DESCRIPTION
**Pull request by jkotas**
`Environment.SystemDirectory` is a lot more efficient than `Environment.GetFolderPath(System.Environment.SpecialFolder.System)`.

Contributes to https://github.com/dotnet/aspnetcore/issues/47269

### CODE REVIEW

File: src/DefaultBuilder/src/WebApplicationBuilder.cs [hunk 1]
```diff
@@ -189,11 +189,11 @@ private static void SetDefaultContentRoot(WebApplicationOptions options, Configu
...
             // value for ContentRoot is AppContext.BaseDirectory (e.g. the executable path) which probably makes more sense than the system32.
 
-            // In my testing, both Environment.CurrentDirectory and Environment.GetFolderPath(Environment.SpecialFolder.System) return the path without
+            // In my testing, both Environment.CurrentDirectory and Environment.SystemDirectory return the path without
             // any trailing directory separator characters. I'm not even sure the casing can ever be different from these APIs, but I think it makes sense to
             // ignore case for Windows path comparisons given the file system is usually (always?) going to be case insensitive for the system path.
             string cwd = System.Environment.CurrentDirectory;
-            if (!OperatingSystem.IsWindows() || !string.Equals(cwd, System.Environment.GetFolderPath(System.Environment.SpecialFolder.System), StringComparison.OrdinalIgnoreCase))
+            if (!OperatingSystem.IsWindows() || !string.Equals(cwd, Environment.SystemDirectory, StringComparison.OrdinalIgnoreCase))
```
Comment (jkotas) [inline src/DefaultBuilder/src/WebApplicationBuilder.cs:196]:
```suggestion
            if (!OperatingSystem.IsWindows() || !string.Equals(cwd, System.Environment.Environment.SystemDirectory, StringComparison.OrdinalIgnoreCase))
```

File: src/DefaultBuilder/src/WebApplicationBuilder.cs [hunk 2]
```diff
@@ -189,11 +189,11 @@ private static void SetDefaultContentRoot(WebApplicationOptions options, Configu
...
             // value for ContentRoot is AppContext.BaseDirectory (e.g. the executable path) which probably makes more sense than the system32.
 
-            // In my testing, both Environment.CurrentDirectory and Environment.GetFolderPath(Environment.SpecialFolder.System) return the path without
+            // In my testing, both Environment.CurrentDirectory and Environment.SystemDirectory return the path without
             // any trailing directory separator characters. I'm not even sure the casing can ever be different from these APIs, but I think it makes sense to
             // ignore case for Windows path comparisons given the file system is usually (always?) going to be case insensitive for the system path.
             string cwd = System.Environment.CurrentDirectory;
-            if (!OperatingSystem.IsWindows() || !string.Equals(cwd, System.Environment.GetFolderPath(System.Environment.SpecialFolder.System), StringComparison.OrdinalIgnoreCase))
+            if (!OperatingSystem.IsWindows() || !string.Equals(cwd, System.Environment.Environment.SystemDirectory, StringComparison.OrdinalIgnoreCase))
```
Comment (jkotas) [inline src/DefaultBuilder/src/WebApplicationBuilder.cs:196]:
```suggestion
            if (!OperatingSystem.IsWindows() || !string.Equals(cwd, System.Environment.SystemDirectory, StringComparison.OrdinalIgnoreCase))
```

**Review by halter73 — APPROVED** [review]

**Review by danmoseley — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]
We will need to update the same code in dotnet/runtime that is linked in the comment in the code.

it would maybe be better to update the runtime first, so we can update the comment in asp.net that has the commit SHA in the URL.

**Pull request merged**